### PR TITLE
handle different verify callbacks

### PR DESCRIPTION
**Pull request by jonstorer**
comply with passport's weirdness

https://github.com/jaredhanson/passport-oauth2/blob/master/lib/strategy.js#L180

### CODE REVIEW

File: lib/passport-mock/strategy.js [hunk 1]
```diff
@@ -40,11 +49,27 @@ MockStrategy.prototype.authenticate = function authenticate (req, options) {
...
         delete this.__proto__._profile;
+        delete this.__proto__._expires_in;
+        delete this.__proto__._scope;
+        delete this.__proto__._token_type;
       };
 
-      this.verify(accessToken, refreshToken, profile, function (err, user) {
-        this.success(user);
-      }.bind(this));
+      var args = [req, accessToken, refreshToken, params, profile, verified];
```
Comment (jonstorer) [inline lib/passport-mock/strategy.js]: rm

File: lib/passport-mock/strategy.js [hunk 2]
```diff
@@ -40,11 +49,24 @@ MockStrategy.prototype.authenticate = function authenticate (req, options) {
...
+      var arity = this.verify.length;
+      if (arity === 6) {
+        this.verify(req, accessToken, refreshToken, params, profile, verified);
+      } else if (arity === 5) {
+        if (this._passReqToCallback) {
+          this.verify(req, accessToken, refreshToken, profile, verified);
+        } else {
+          this.verify(accessToken, refreshToken, params, profile, verified);
+        };
+      } else { // arity === 4
```
Comment (tripleentendre) [inline lib/passport-mock/strategy.js]: Would it make sense to change this to else if and have the else block throw an exception if the length is something cray?

**Comment by ekump** [comment]
LGTM

File: lib/passport-mock/strategy.js [hunk 3]
```diff
@@ -40,11 +49,24 @@ MockStrategy.prototype.authenticate = function authenticate (req, options) {
...
+        delete this.__proto__._expires_in;
+        delete this.__proto__._scope;
+        delete this.__proto__._token_type;
       };
 
-      this.verify(accessToken, refreshToken, profile, function (err, user) {
-        this.success(user);
-      }.bind(this));
+      var arity = this.verify.length;
+      if (arity === 6) {
```
Comment (kylessnell) [inline lib/passport-mock/strategy.js]: this if statement feels pretty gross.

**Comment by kylessnell** [comment]
lgtm otherwise

File: lib/passport-mock/strategy.js (hunk 2 above)
Comment (jonstorer) [inline lib/passport-mock/strategy.js]: I could, but I'm not sure of it's value... if you look at the lib that I'm mimicing, he didn't do it either

**Pull request closed without merging**